Playlist Use-Case Added

saadalish opened the pull request:
Following files are changed or added in this commit:
1) playlists/urls.py , new urls added related to playlist like create, edit or delete a playlist
2) playlist/views.py , new view functions added, like create, add_song_to_playlist
3) playlists/forms.py , new form created for playlist
4) playlist/templates , Html templates created for create, update, delete a playlist

saadalish committed:
Playlist Use Case
Following files are changed or added in this commit:
1) playlists/urls.py , new urls added related to playlist like create, edit or delete a playlist
2) playlist/views.py , new view functions added, like create, add_song_to_playlist
3) playlists/forms.py ,new form created for playlist
4) playlist/templates , Html templates created for create, update, delete a playlist

sheralim012 commented on playlists/urls.py:
```diff
@@ -1 +1,12 @@
+    path('create', views.create, name='create'),
+    path('<playlist_id>', views.update, name='update'),
+    path('<playlist_id>/delete', views.delete, name='delete'),
+    path('<playlist>/<song>/add', views.add_song, name='AddSongPlaylist'),
```
use consistent naming for urls

sheralim012 replied:
use snake case

sheralim012 commented on playlists/views.py:
```diff
@@ -1,2 +1,64 @@
+    if request.method == 'POST':
+        form = PlaylistForm(request.POST)
+        if form.is_valid():
+            form.instance.user = request.user
```
could you double check this

sheralim012 commented on playlists/views.py:
```diff
@@ -1,2 +1,64 @@
+    playlist = get_object_or_404(Playlist, id=playlist_id)
+    form = PlaylistForm(request.POST or None, instance=playlist)
+    songs_in_playlist = SongPlaylist.objects.filter(playlist__exact=playlist).values_list('song_id', flat=True)
+    recommended_songs = Song.objects.exclude(id__in=songs_in_playlist)
```
remove this for now

sheralim012 commented on playlists/views.py:
```diff
@@ -1,2 +1,64 @@
+def update(request, playlist_id):
+    playlist = get_object_or_404(Playlist, id=playlist_id)
+    form = PlaylistForm(request.POST or None, instance=playlist)
+    songs_in_playlist = SongPlaylist.objects.filter(playlist__exact=playlist).values_list('song_id', flat=True)
```
remove __exact

sheralim012 commented on playlists/views.py:
```diff
@@ -1,2 +1,64 @@
+
+
+def update(request, playlist_id):
+    playlist = get_object_or_404(Playlist, id=playlist_id)
```
do R&D for releations

sheralim012 commented on playlists/views.py:
```diff
@@ -1,2 +1,64 @@
+def delete(request, playlist_id):
+    playlist = get_object_or_404(Playlist, id=playlist_id)
+    playlist.delete()
+    return HttpResponseRedirect("/playlists")
```
use url names

sheralim012 commented on playlists/templates/playlists/index.html:
```diff
@@ -0,0 +1,18 @@
+                <a href="{% url 'update' playlist.id %}">{{ playlist.name }}</a>
+            <hr/>
+            {% endfor %}
+        {% endif %}
```
and else case when no playlist is available

saadalish committed:
Resolved - request issues

saadalish committed:
models and views updated
playlist, song and queue models updated
views updated according to new model fields
urls updated, camelcase removed

saadalish committed:
Migration files for updated models

sheralim012 reviewed (changes requested):
why migrations are deleted?

saadalish committed:
Migrations Added and a typo fixed
